packages/compose/src/utils/debounce/index.ts

```diff
@@ -0,0 +1,221 @@
+export interface DebounceOptions {
+	leading: boolean;
+	maxWait: number;
+	trailing: boolean;
+}
+
+export interface DebouncedFunc< T extends ( ...args: any[] ) => any > {
+	/**
+	 * Call the original function, but applying the debounce rules.
+	 *
+	 * If the debounced function can be run immediately, this calls it and returns its return
+	 * value.
+	 *
+	 * Otherwise, it returns the return value of the last invocation, or undefined if the debounced
+	 * function was not invoked yet.
+	 */
+	( ...args: Parameters< T > ): ReturnType< T > | undefined;
+
+	/**
+	 * Throw away any pending invocation of the debounced function.
+	 */
+	cancel(): void;
+
+	/**
+	 * If there is a pending invocation of the debounced function, invoke it immediately and return
+	 * its return value.
+	 *
+	 * Otherwise, return the value from the last invocation, or undefined if the debounced function
+	 * was never invoked.
+	 */
+	flush(): ReturnType< T > | undefined;
+}
+
+/**
+ * A simplified and properly typed version of lodash's `debounce`, that
+ * always uses timers instead of sometimes using rAF.
+ *
+ * Creates a debounced function that delays invoking `func` until after `wait`
+ * milliseconds have elapsed since the last time the debounced function was
+ * invoked, or until the next browser frame is drawn. The debounced function
```

Comment:
Looks like `, or until the next browser frame is drawn` unintentionally survived the `rAF` purge. We should probably remove it, as there's nothing in this function that is related to the browser rendering pipeline now.